### PR TITLE
FDB-Flush-API

### PR DESCRIPTION
**Pull request by arunsubash**
Consolidation of FDB Flush API's. Provide options to flush Static/Dynamic/All entries

### CODE REVIEW

File: sai/inc/saifdb.h [hunk 1]
```diff
@@ -102,6 +102,52 @@ typedef enum _sai_fdb_entry_attr_t
...
+* Additionally the type of entry needs to be flushed can be specified
+  using SAI_FDB_FLUSH_ATTR_ENTRY_TYPE which can be static or dynamic or all
+*/
+typedef enum _sai_fdb_flush_attr {
+
+   /*Flush based all entries. No value required*/
+   SAI_FDB_FLUSH_ATTR_ALL,
+
+   /*Flush based on MAC [sai_mac_t]*/
+   SAI_FDB_FLUSH_ATTR_MAC,
```
Comment (lguohan) [inline sai/inc/saifdb.h]: do we need this one? this does not appear in the comment.

File: sai/inc/saifdb.h [hunk 2]
```diff
@@ -102,6 +102,52 @@ typedef enum _sai_fdb_entry_attr_t
...
+
+   /*Flush based on port [sai_object_id_t]*/
+   SAI_FDB_FLUSH_ATTR_PORT_ID,
+
+   /*Flush based on VLAN [sai_vlan_id_t]*/
+   SAI_FDB_FLUSH_ATTR_VLAN_ID,
+
+   /*Flush based on entry type [sai_fdb_flush_entry_type_t]*/
+   SAI_FDB_FLUSH_ATTR_ENTRY_TYPE,
```
Comment (lguohan) [inline sai/inc/saifdb.h]: can we re-use the sai_fdb_entry_type_t here? If you do not provide this attribute, then it means all types of entry. Seems to you do not need SAI_FDB_FLUSH_ENTRY_ALL and you can reuse sai_fdb_entry_type_t.

Comment (arunsubash) [inline sai/inc/saifdb.h]: Yes we can reuse the sai_fdb_entry_type_t. Deleted the MAC and ALL option from sai_fdb_flush_attr and updated the pull request

**Comment by lguohan** [comment]
merged

**Pull request closed without merging**